frontend/src/services/PlanService.ts

```diff
@@ -218,6 +219,65 @@ export const fetchComments = (planId: number, studentId: number) =>
     })
   );
 
+/* 
+    Adds a full year to the current plan
+  */
+export const addYear = (
+  userId: number,
+  planId: number,
+  plan: Partial<IUpdatePlanData>
+) => {
+  // Logic here is to simulate backend changing the plan
+  const schedule: DNDSchedule = plan.schedule!;
+  const currYear: number = schedule.years[schedule.years.length - 1] + 1!;
+  const years = [...schedule.years, currYear];
+  const yearMap = {
+    ...schedule.yearMap,
+    [currYear]: {
+      year: currYear,
+      fall: {
+        season: "FL",
+        year: currYear,
+        termId: currYear * 100 + 10,
+        status: "CLASSES",
+        classes: [],
+      },
+      isSummerFull: false,
+      spring: {
+        season: "SP",
+        year: currYear,
+        termId: currYear * 100 + 30,
+        status: "INACTIVE",
+        classes: [],
+      },
+      summer1: {
+        season: "S1",
+        year: currYear,
+        termId: currYear * 100 + 40,
+        status: "INACTIVE",
+        classes: [],
+      },
+      summer2: {
+        season: "S2",
+        year: currYear,
+        termId: currYear * 100 + 60,
+        status: "INACTIVE",
+        classes: [],
+      },
+    },
+  };
+  const newPlan = { ...plan, schedule: { ...schedule, years, yearMap } };
+
+  return fetch(`/api/users/${userId}/plans/${planId}`, {
```

Comment:
Do we have axios as a dependency? If so, use that over fetch.

Reply:
we don't have axios as a dependency :(